### PR TITLE
(#554) Fixes to get couple of tests running cleanly in ASAN-builds

### PR DESCRIPTION
**Pull request by gapisback**
This commit fixes minor errors in 2 tests (io_apis_test, filter_test) to get them running cleanly in ASAN-builds.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -59,6 +59,13 @@ function usage() {
    echo "To run CI-regression tests       : INCLUDE_SLOW_TESTS=true ./${Me}"
    echo "To run nightly regression tests  : RUN_NIGHTLY_TESTS=true ./${Me}"
    echo "To run make build-and-test tests : RUN_MAKE_TESTS=true ./${Me}"
+   echo
+   echo "To run a smaller collection of slow running tests,
+name the function that drives the test execution.
+Examples:"
+   echo "  INCLUDE_SLOW_TESTS=true ./test.sh run_btree_tests"
+   echo "  INCLUDE_SLOW_TESTS=true ./test.sh run_splinter_functionality_tests"
+   echo "  INCLUDE_SLOW_TESTS=true ./test.sh nightly_cache_perf_tests"
```
Comment (gapisback) [inline test.sh:68]: Slightly unrelated change. I needed a way to quickly re-test batches of tests run by this script under `ASAN` build modes. So, added this new capability to run a set of tests w/o having to wait for full runs to complete to validate a fix.

File: test.sh [hunk 2]
```diff
@@ -743,6 +750,24 @@ fi
...
+# function takes arguments, pass them on the command-line. This way, one
+# can debug script changes to ensure that test-execution still works.
+#
+# Examples:
+#      INCLUDE_SLOW_TESTS=true ./test.sh run_btree_tests
+# ------------------------------------------------------------------------
+if [ $# -ge 1 ]; then
+
+   # shellcheck disable=SC2048
+   $*
```
Comment (gapisback) [inline test.sh:765]: This is where the support to re-run manually granular collection of tests is added. This will help expedite dev turnaround during fix stabilization and re-tests.

**Comment by gapisback** [comment]
With these two fixes, my slow re-run of `test.sh` passes clean in ASAN build-mode. (MSAN build-mode tests are still running.)

```
638 Wed Mar 22 12:26:30 PDT 2023 **** SplinterDB Test Suite Execution Times ****
639
640 Fast unit tests                                                       :  124s [  0h  2m  4s ]
641 Splinter inserts test                                                 :  249s [  0h  4m  9s ]
642 Splinter lookups test                                                 :  177s [  0h  2m 57s ]
643 Splinter print diagnostics test                                       :  254s [  0h  4m 14s ]
644 Functionality test, key size=8 bytes                                  :   38s [  0h  0m 38s ]
645 Functionality test, with default key size                             :   36s [  0h  0m 36s ]
646 Functionality test, default key size, with background threads         :   35s [  0h  0m 35s ]
647 Functionality test, key size=maximum (102 bytes)                      :   42s [  0h  0m 42s ]
648 Very quick Performance test                                           :    5s [  0h  0m  5s ]
649 Quick Performance test with bg-threads                                :    5s [  0h  0m  5s ]
650 Performance test                                                      :  321s [  0h  5m 21s ]
651 BTree test, key size=8 bytes                                          : 1173s [  0h 19m 33s ]
652 BTree test, with default key size                                     :  981s [  0h 16m 21s ]
653 BTree test, key size=100 bytes                                        :  647s [  0h 10m 47s ]
654 BTree Perf test                                                       :    0s [  0h  0m  0s ]
655 Cache test                                                            :   11s [  0h  0m 11s ]
656 Log test                                                              :    1s [  0h  0m  1s ]
657 Filter test                                                           :   11s [  0h  0m 11s ]
658 All Tests                                                             : 4397s [  1h 13m 17s ]
```

**Comment by gapisback** [comment]
Slow running tests from `test.sh` in MSAN build-mode are also running clean now:

```
File: ~/tmp/test.rel-msan.slow.out

637 ALL PASSED
638 Wed Mar 22 13:46:20 PDT 2023 **** SplinterDB Test Suite Execution Times ****
639
640 Fast unit tests                                                       :   66s [  0h  1m  6s ]
641 Splinter inserts test                                                 :  105s [  0h  1m 45s ]
642 Splinter lookups test                                                 :   96s [  0h  1m 36s ]
643 Splinter print diagnostics test                                       :  107s [  0h  1m 47s ]
644 Functionality test, key size=8 bytes                                  :   14s [  0h  0m 14s ]
645 Functionality test, with default key size                             :   14s [  0h  0m 14s ]
646 Functionality test, default key size, with background threads         :   14s [  0h  0m 14s ]
647 Functionality test, key size=maximum (102 bytes)                      :   16s [  0h  0m 16s ]
648 Very quick Performance test                                           :    2s [  0h  0m  2s ]
649 Quick Performance test with bg-threads                                :    1s [  0h  0m  1s ]
650 Performance test                                                      :  162s [  0h  2m 42s ]
651 BTree test, key size=8 bytes                                          :  401s [  0h  6m 41s ]
652 BTree test, with default key size                                     :  353s [  0h  5m 53s ]
653 BTree test, key size=100 bytes                                        :  206s [  0h  3m 26s ]
654 BTree Perf test                                                       :    0s [  0h  0m  0s ]
655 Cache test                                                            :    9s [  0h  0m  9s ]
656 Log test                                                              :    1s [  0h  0m  1s ]
657 Filter test                                                           :    5s [  0h  0m  5s ]
658 All Tests                                                             : 1694s [  0h 28m 14s ]
659
660 Wed Mar 22 14:14:34 PDT 2023 End SplinterDB Test Suite Execution.
```

**Review by rosenhouse — APPROVED** [review]
Seems fine
